pkg/apis/config/v1alpha1/defaults.go

```diff
@@ -21,3 +21,16 @@ import (
 func addDefaultingFuncs(scheme *runtime.Scheme) error {
 	return RegisterDefaults(scheme)
 }
+
+// SetDefaults_ControllerConfiguration sets default values for ControllerConfiguration objects.
+func SetDefaults_ControllerConfiguration(obj *ControllerConfiguration) {
```

Comment:
Hm, this type is not served by API, it is used just to parse CRDs, so I am not sure that this defaulting will be applied.

Reply:
I think we have to explicitly call it, similar to https://github.com/gardener/gardener/blob/master/cmd/gardener-controller-manager/app/gardener_controller_manager.go#L129-L143, but I can't find it in https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/cmd/config.go#L28:6.

Reply:
I've updated the PR - now decoding happens via the `UniversalDecoder().Decode(...)`  method of the codex factory. Defaulting happens automatically. See `pkg/apis/config/loader/loader_test.go` - there is a test which tests exactly this behavior.